### PR TITLE
format: fix for nvme

### LINKED ISSUE #2844: mount: not working on nvme
**Description**

**Steps to reproduce the issue:**
boot on a system with an empty nvme attached and 
```
init:
- linuxkit/init:26def2174c74efa0ea8006ebd63ebb5d02e9513e
- linuxkit/runc:abc3f292653e64a2fd488e9675ace19a55ec7023
- linuxkit/containerd:e58a382c33bb509ba3e0e8170dfaa5a100504c5b
- linuxkit/ca-certificates:de21b84d9b055ad9dcecc57965b654a7a24ef8e0
- linuxkit/getty:22e27189b6b354e1d5d38fc0536a5af3f2adb79f
onboot:
- name: sysctl
  image: linuxkit/sysctl:4c1ef93bb5eb1a877318db4b2daa6768ed002e21
- name: sysfs
  image: linuxkit/sysfs:1284b4a7061a5cc426425f0fb00748192505a05f
- name: format
  image: linuxkit/format:2496b87d7923cb8f30247962a03fff023e95db2d
- name: mounts
  image: linuxkit/mount:b346ec277b7074e5c9986128a879c10a1d18742b-dirty
  command:
  - /usr/bin/mountie
  - /var/lib

```

**Describe the results you received:**
```
Failed to find block device /dev/nvme0n11 - stat returned: stat /dev/nvme0n11: no such file or directory
```

**Describe the results you expected:**
mounted fs

**Additional information you deem important (e.g. issue happens only occasionally):**
the nvme device is called `nvme0n1` and the created partition from the format container `nvme0n1p1`

### PR DESCRIPTION
**Pull request by w9n**
**- What I did**
fix #2844
**- How I did it**
check if last char of partition is numeric
**- How to verify it**
run on nvme
**- Description for the changelog**

mount: fix for nvme naming convention

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: pkg/format/format.go [hunk 1]
```diff
@@ -171,7 +172,14 @@ func format(d, label, fsType string, forced bool) error {
 		return fmt.Errorf("Error running blockdev: %v", err)
 	}
 
-	partition := fmt.Sprintf("%s1", d)
+	// check if last char is numeric
+	var partition string
+	if _, err := strconv.Atoi(d[len(d)-1:]); err == nil {
```
Comment (justincormack) [inline pkg/format/format.go]: I think its easier to check if it is `> '0'` and `< '9'`

Comment (w9n) [reply pkg/format/format.go]: fixed

**Comment by justincormack** [comment]
LGTM. We need to build a new format package.

**Comment by rn** [comment]
I've build/pushed the updated package. Could you, in a separate commit update the hashes in the YAML files to `linuxkit/format:5a1140cb65e733f26de727fa684fef1326e9d5ab` There is a script in `./scripts/update-component-sha.sh` for it

**Comment by w9n** [comment]
almost forgot about these hashes ;) done.

**Pull request merged**